src/hooks/use-search/use-search.ts

```diff
@@ -0,0 +1,247 @@
+import { Dispatch, useEffect, useReducer } from 'react';
+// @ts-ignore - addsearch has no types
+import AddSearchClient from 'addsearch-js-client';
+
+// NOTE: all the types are documented here https://www.addsearch.com/docs/api/search-results/
+type Image = {
+  main: string;
+  main_b64: string;
+  capture: string;
+};
+
+type DocumentType = 'html' | 'pdf' | 'doc' | 'docx' | 'ppt' | 'pptx';
+
+type Hit = {
+  id: string;
+  url: string;
+  title: string;
+  meta_description: string;
+  meta_categories: string[]; // <meta name="addsearch-category" content="features/api" />
+  custom_fields: Record<string, string> | Record<string, string[]>;
+  highlight: string; // NOTE: comes with html tags like <em>, etc.
+  ts: string; // NOTE: essentially an ISO date string
+  categories: string[];
+  document_type: DocumentType;
+  images: Image;
+  score: number; // NOTE: a number between 0 and 1
+};
+
+type Result = {
+  error?: { message: string } | null;
+  page: number;
+  total_hits: number;
+  processing_time_ms: number;
+  hits: Hit[];
+  facets: null;
+  rangeFacets: null;
+};
+
+export type State = {
+  client: string | null;
+  query?: string;
```

Comment:
query, by design, should be always a string (empty one by default) - was there code that made it not so?

Reply:
yep, my bad 👍